src/ar_node_utils.erl

```diff
@@ -35,19 +35,26 @@ get_conflicting_txs(StateTXs, TX) ->
 			)
 	].
 
-%% @doc Get a specific full block (a block containing full txs) via
-%% blocks indep_hash.
+%% @doc Get a full block (a block containing all transactions) by the independent hash.
+%%      Try to find the block locally first. If we do not have the full block on disk, try to download it from peers.
 get_full_block(Peers, ID, BHL) when is_list(Peers) ->
-	% check locally first, if not found ask list of external peers for block
+	GetBlockFromPeersFun = fun() ->
+		get_full_block_from_remote_peers(ar_util:unique(Peers), ID, BHL)
+	end,
 	case ar_storage:read_block(ID, BHL) of
 		unavailable ->
-			get_full_block_from_remote_peers(ar_util:unique(Peers), ID, BHL);
+			GetBlockFromPeersFun();
 		Block ->
 			case make_full_block(Block) of
-				unavailable ->
-					ar_storage:invalidate_block(Block),
-					get_full_block(Peers, ID, BHL);
-				FinalB ->
+				{error, unavailable} ->
+					GetBlockFromPeersFun();
+				{error, {txs_missing, MissingTXIDs}} ->
+					ar:info([
+						{transactions_missing_on_disk_for_block, ar_util:encode(ID)},
+						{missing_txs, lists:map(fun ar_util:encode/1, MissingTXIDs)}
+					]),
+					GetBlockFromPeersFun();
+				{ok, FinalB} ->
```

Comment:
6 lines further down (I cannot make a comment there), this function returns `make_full_block(ID, BHL)`. We need to case it to return the block or `unavailable` instead of the ok or error tuple.

Reply:
✅

Reply:
Yes, looks good. I'm putting my ✅on it too 😉